setup.py

```diff
@@ -214,7 +252,62 @@ def monkey_patch_THD_link_flags():
     C.extra_link_args += thd_deps
 
 
-class build_ext(setuptools.command.build_ext.build_ext):
+class ninja_build_ext(setuptools.command.build_ext.build_ext):
```

Comment:
Could we e.g. put the ninja code somewhere in `tools/`? That would make `setup.py` more readable